src/HotelCalifornia/Exec.hs

```diff
@@ -38,15 +39,18 @@ parseExecArgs = do
 runExecArgs :: ExecArgs -> IO ()
 runExecArgs ExecArgs {..} = do
     let script =
-            unwords $ NEL.toList execArgsScript
+            Escape.escapeMany $ NEL.toList execArgsScript
         spanName =
             fromMaybe (Text.pack script) execArgsSpanName
 
     inSpan' spanName \span_ -> do
         newEnv <- spanContextToEnvironment span_
         fullEnv <- mappend newEnv <$> getEnvironment
 
-        let processConfig = shell $ unwords $ NEL.toList execArgsScript
+        let command :| args = execArgsScript
+
+        let processConfig = proc command args
```

Comment:
I was wondering about this distinction! From what I coudl tell, `proc` looks up the executable, and then Does The Thing. Whereas `shell` passes the input text to `sh` or something.

The intent is to let people write stuff that's more like a shell script as the input. Hypothetically I'm imagining something like `hotel exec -- foo && bar`, where `proc` would see this as `proc "foo" ["&&", "bar"]`, but `shell` would get `shell "foo && bar"`.

But I'll grant I may be *way* off and it'd actually be parsed as `(hotel exec foo) && bar`

Reply:
Yeah, so the way to get hat I was thinking of is a bit gnarly: `hotel exec -- 'echo a && echo b'`. Which does work currently but not with this patch.

Testing on the work codebase, this patch does fix a bug (thanks for anticipating it!) and all commands work like I want them to, so I'm inclined to accept as-is. Possibly a flag like `--raw-shell` would interpet the `SCRIPT` argument different and pass it in with `shell` instead of `proc`?

Reply:
I can add support for shell commands, too